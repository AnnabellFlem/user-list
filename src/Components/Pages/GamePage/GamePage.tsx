import React, { useState } from 'react'
import MainLayout from './MainLayout'
import Header from './Header'
import Footer from './Footer'

type GamePage = {
  handleEndGame: (isEndGame: boolean) => void
  handleQuestion: (question: number) => void
  currentNumberQuestion: number
}

const GamePage: React.FC<GamePage> = ({
  handleEndGame,
  handleQuestion,
  currentNumberQuestion,
}) => {
  const [openFavList, setOpenFavList] = useState(false)

  const handleBtnClick = () => {
    setOpenFavList(!openFavList)
  }

  return (
    <>
      <Header
        handleBtnClick={handleBtnClick}
        currentNumberQuestion={currentNumberQuestion}
      />
      <MainLayout
        handleCurrentQuestion={handleQuestion}
        currentNumberQuestion={currentNumberQuestion}
        handleEndGame={handleEndGame}
      />
      <Footer />
    </>
  )
}

export default GamePage
