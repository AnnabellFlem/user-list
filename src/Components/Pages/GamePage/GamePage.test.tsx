import React from 'react'
import { mount } from 'enzyme'
import GamePage from './GamePage'

const MOCK_HANDLE_END_GAME = jest.fn()
const MOCK_HANDLE_QUESTION = jest.fn()

describe('<GamePage /> test', () => {
  let wrapper: any

  beforeAll(() => {
    wrapper = mount(
      <GamePage
        currentNumberQuestion={0}
        handleEndGame={MOCK_HANDLE_END_GAME}
        handleQuestion={MOCK_HANDLE_QUESTION}
      />,
    )
  })

  it('should to match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Component should render without crashing', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('Should render header', () => {
    expect(wrapper.find('header')).toHaveLength(1)
  })

  it('Should render main', () => {
    expect(wrapper.find('main')).toHaveLength(1)
  })

  it('Should render footer', () => {
    expect(wrapper.find('footer')).toHaveLength(1)
  })
})
