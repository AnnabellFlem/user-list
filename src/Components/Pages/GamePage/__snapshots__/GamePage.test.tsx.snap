// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GamePage /> test should to match snapshot 1`] = `ReactWrapper {}`;
